### PR TITLE
[client/rest]: refactor most of process hosting into server/process

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: rosetta is bundled within catapult REST directly
solution: move rosetta implementation into new mini REST

### CODE REVIEW

File: client/rest/resources/rest.rosetta.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "port": 4000,
+  "protocol": "HTTPS",
+  "sslKeyPath": "",
+  "sslCertificatePath": "",
+  "crossDomain": {
+    "allowedHosts": ["*"],
+    "allowedMethods": ["GET", "POST", "PUT", "OPTIONS"]
+  },
+
+  "restEndpoint": "http://localhost:3000",
```
Comment (Wayonb) [inline client/rest/resources/rest.rosetta.json:16]: I am not sure if it's possible to call the HTTP endpoint from HTTPS.

File: client/rest/src/index.js [hunk 2]
```diff
@@ -122,10 +59,7 @@ const registerRoutes = (server, db, services) => {
 	const servicesView = {
 		config: {
 			network: services.config.network,
-			rest: {
-				protocol: services.config.protocol,
-				port: services.config.port
-			},
+			restEndpoint: `${services.config.protocol}://localhost:${services.config.port}`,
```
Comment (Wayonb) [inline client/rest/src/index.js:62]: Should this be restEndpoint in the config?

**Review by Wayonb — APPROVED** [review]

File: client/rest/src/index.js (hunk 2 above)
Comment (Jaguar0625) [reply client/rest/src/index.js:62]: this is only when rosetta is hosted within main REST, so, this is just endpoint of REST (which is shared with rosetta REST)

File: client/rest/resources/rest.rosetta.json (hunk 1 above)
Comment (Jaguar0625) [reply client/rest/resources/rest.rosetta.json:16]: in that case, we'd need to update it as part of deployment

**Pull request merged**